pynamodb/tests/integration/test_transaction_integration.py

```diff
@@ -0,0 +1,257 @@
+        User(2).save(in_transaction=transaction)
+
+        try:
+            # committing the first time, then adding more info and committing again
+            transaction.commit()
+            User(3).save(in_transaction=transaction)
+            transaction.commit()
+            assert False, 'Failed to raise error'
+        except PutError as e:
+            assert get_error_code(e) == IDEMPOTENT_PARAMETER_MISMATCH
+
+    def test_transact_write__error__transaction_cancelled(self):
+        # create a users and a bank statements for them
+        User(1).save()
+        BankStatement(1).save()
+
+        # attempt to do this as a transaction with the condition that they don't already exist
+        transaction = TransactWrite()
+        User(1).save(condition=(User.user_id.does_not_exist()), in_transaction=transaction)
+        BankStatement(1).save(condition=(BankStatement.user_id.does_not_exist()), in_transaction=transaction)
+        try:
+            transaction.commit()
+            assert False, 'Failed to raise error'
+        except PutError as e:
+            assert get_error_code(e) == TRANSACTION_CANCELLED
+
+    # def test_transact_write__error__transaction_in_progress(self):
+    #     transact_get = TransactGet()
+    #     User.get(1, in_transaction=transact_get)
+    #     BankStatement.get(1, in_transaction=transact_get)
+    #
+    #     transact_write = TransactWrite()
+    #     User(1).save(in_transaction=transact_write)
+    #     BankStatement(1).save(in_transaction=transact_write)
+    #
+    #     try:
+    #         # getting and writing to the same models at the same time
+    #         transact_write.commit()
+    #         transact_get.commit()
+    #         assert False, 'Failed to raise error'
+    #     except (GetError, PutError) as e:
+    #         assert get_error_code(e) == TRANSACTION_IN_PROGRESS
+
+    def test_transact_get(self):
+        # making sure these entries exist, and with the expected info
+        User(1).save()
+        BankStatement(1).save()
+        User(2).save()
+        BankStatement(2, balance=100).save()
+
+        # get users and statements we just created
+        transaction = TransactGet()
+        User.get(1, in_transaction=transaction)
+        BankStatement.get(1, in_transaction=transaction)
+        User.get(2, in_transaction=transaction)
+        BankStatement.get(2, in_transaction=transaction)
+        transaction.commit()
+
+        # assign them to variables after commit
+        user1 = transaction.from_results(User, 1)
+        user2 = transaction.from_results(User, 2)
+        statement1 = transaction.from_results(BankStatement, 1)
+        statement2 = transaction.from_results(BankStatement, 2)
+
+        assert user1.user_id == 1
+        assert statement1.user_id == 1
+        assert statement1.balance == 0
+        assert user2.user_id == 2
+        assert statement2.user_id == 2
+        assert statement2.balance == 100
```

Comment:
nit: could have common func that ensures user/bank/statement models are equivalent